share/functions/help.fish

```diff
@@ -136,13 +136,13 @@ function help --description 'Show help for the fish shell'
         # Help is installed, use it
         set page_url file://$__fish_help_dir/$fish_help_page
 
-        # For Windows (Cygwin and WSL), we need to convert the base help dir to a Windows path before converting it to a file URL
+        # For Windows (Cygwin, msys2 and WSL), we need to convert the base help dir to a Windows path before converting it to a file URL
         # but only if a Windows browser is being used
         if type -q cygpath
-            and string match -qr cygstart $fish_browser[1]
+            and string match -qr 'cygstart|cmd\.exe$' $fish_browser[1]
             set page_url file://(cygpath -m $__fish_help_dir)/$fish_help_page
         else if type -q wslpath
-            and string match -qr '.exe' $fish_browser[1]
+            and string match -qr 'cmd\.exe$' $fish_browser[1]
```

Comment:
I'm not sure if this needs to be done *only* for cmd.exe?

Reply:
Indeed, I was thinking about `\.exe$` but in a WSL environment there might be programs without the `.exe` extension. So this check might be superfluous ?

Reply:
The check makes sense as far as I can tell: if the browser is a non-Windows executable, like `w3m` instead of `firefox.exe`, then we don't want to convert the path.

Reply:
I see, I'll switch this to `\.exe$` tommorow. (I know there is a similar match a little further, I'll change that too).

Reply:
I think `\.exe *` might be better since it allows `$fish_browser` to contains both `browser.exe` and `browser.exe --with-arguments`. What do you think ?

Reply:
I believe so. `\.exe\b` comes to mind.

Reply:
That would match `some.exe--wrong-arg` without the whitespace after exe.
I don't know if that's a problem or not, maybe I'm going too far. Otherwise `\.exe(\s+|$)` seems good.